### PR TITLE
Add Makefile, use golangci-lint, move to GitHub actions

### PR DESCRIPTION
**Pull request by guggero**
As a new contributor to the Neutrino repository it was a bit of a challenge to figure out how to use the clunky `gotest.sh` file. That's why I think it would be nice to switch over to a `Makefile`.

At the same time I felt like we should fix a number of smaller issues that an updated linter should find. That's why I opted to add the `golangci-lint` linter as well.
And I think I might even have discovered a possible explanation for some of the flakes we see in the `lnd` itests with Neutrino (see 0a824919, could also be mistaken about the significance of this).

### CODE REVIEW

File: blockntfns/manager.go
Comment (wpaulino) [inline blockntfns/manager.go:142]: Nit: could also provide it as an argument to the goroutine func.

File: batch_spend_reporter.go
Comment (wpaulino) [inline batch_spend_reporter.go:73]: :+1:

File: blockmanager.go
Comment (wpaulino) [inline blockmanager.go:806]: Nit: replace `!` with a comma?

File: neutrino.go
Comment (wpaulino) [inline neutrino.go:1492]: :+1:

File: blockmanager_test.go
Comment (wpaulino) [inline blockmanager_test.go:443]: Can we remove the lint exception for all the `t.Fatal` in goroutines and follow https://golang.org/doc/go1.16#vet?

File: headerfs/store.go
Comment (wpaulino) [inline headerfs/store.go:95]: Interesting. I removed the lint exception to see what the warning was and got `mtx is unused`. It seems to not like the fact that `headerStore` is embedded in other structs.

File: neutrino.go
Comment (wpaulino) [inline neutrino.go:161]: The ban scoring logic was removed in a previous commit, this should probably be removed there as well.

File: pushtx/broadcaster.go
Comment (wpaulino) [inline pushtx/broadcaster.go:84]: :+1:

File: query/workmanager_test.go
Comment (wpaulino) [inline query/workmanager_test.go:211]: Looks like we can remove the `select` since we're only waiting on one channel?

**Review by wpaulino — COMMENTED** [review]
Neutrino was clearly in need of a better linter! `btcwallet` would be another good candidate.

It looks like the GitHub actions aren't running yet, is there anything else that needs to be done?

File: neutrino.go
Comment (guggero) [reply neutrino.go:161]: Right, removed.

File: blockmanager_test.go
Comment (guggero) [reply blockmanager_test.go:443]: Ah, I didn't know it was that easy to fix (expected a bigger refactor). Thanks for the hint, done!

**Comment by guggero** [comment]
> It looks like the GitHub actions aren't running yet, is there anything else that needs to be done?

Changes in GH actions in a PR aren't applied until the PR is merged. I think it's a security measure to make sure repository secrets cannot be extracted easily just by opening a PR.

File: headerfs/store.go
Comment (guggero) [reply headerfs/store.go]: Yeah, this is a bit annoying. I tried the newest version of `golanglint-ci` but that also still flags it. Added a comment to make clear why it's a false positive.

**Review by wpaulino — APPROVED** [review]

**Comment by guggero** [comment]
@cfromknecht PTAL

File: .golangci.yml
Comment (halseth) [inline .golangci.yml:24]: maybe some of these could be enabled?

File: batch_spend_reporter.go
Comment (halseth) [inline batch_spend_reporter.go:64]: yeah, good to catch these! 😀

**Review by halseth — APPROVED** [review]
Excellent! Awesome to finally bring neutrino into the new decade 🚀

**Comment by halseth** [comment]
@guggero Feel free to merge when you've checked out whether the comments are actionable

File: .golangci.yml
Comment (guggero) [reply .golangci.yml]: Yeah, why not. I enabled `malign` globally but only fixed one struct that is used often. For the others the readability/order of the struct would suffer too much for very little benefit.

Also skipping the `prealloc` check only for test code.

**Comment by guggero** [comment]
@halseth I rebased and added two more commits. I don't have merge permissions in this repo so feel free to push the button if you think this is ready.

File: .golangci.yml
Comment (halseth) [inline .golangci.yml:50]: 👍

**Review by halseth — APPROVED** [review]

**Pull request merged**
